File: burgess/src/main/java/com/uncle2000/lib/uitils/T.java
package com.uncle2000.lib.uitils;

import android.os.Looper;
import android.view.Gravity;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;


/**
 * Toast统一管理类
 */
public class T {

    public static boolean isShow = true;

    private T() {
        /* cannot be instantiated */
        throw new UnsupportedOperationException("cannot be instantiated");
    }

    /**
     * 短时间显示Toast
     *
     * @param message
     */
    public static void showShort(CharSequence message) {
//        try {
//            if (isShow && App.INSTANCE != null) {
//
//                Toast toast = Toast.makeText(App.INSTANCE, message, Toast.LENGTH_SHORT);
//                //处理换行文字居中问题
//                try {
//                    ViewGroup view = (ViewGroup) toast.getView();
//                    if (view.getChildCount() > 0) {
//                        TextView textView = (TextView) view.getChildAt(0);
//                        textView.setGravity(Gravity.CENTER);
//                    }
//
//                } catch (Exception e) {
//                    e.printStackTrace();
//                }
//
//                toast.show();
//            }
//        } catch (Exception e) {
//            //解决在子线程中调用Toast的异常情况处理
//            Looper.prepare();
//            Toast.makeText(App.INSTANCE, message, Toast.LENGTH_SHORT).show();
//            Looper.loop();
//        }
    }

    /**
     * 短时间显示Toast
     *
     * @param message
     */
    public static void showShort(int message) {
//        if (isShow)
//            Toast.makeText(App.INSTANCE, message, Toast.LENGTH_SHORT).show();
    }

    /**
     * 长时间显示Toast
     *
     * @param message
     */
    public static void showLong(CharSequence message) {
//        if (isShow)
//            Toast.makeText(App.INSTANCE, message, Toast.LENGTH_LONG).show();
    }

    /**
     * 长时间显示Toast
     *
     * @param message
     */
    public static void showLong(int message) {
//        if (isShow)
//            Toast.makeText(App.INSTANCE, message, Toast.LENGTH_LONG).show();
    }

    /**
     * 自定义显示Toast时间
     *
     * @param message
     * @param duration
     */
    public static void show(CharSequence message, int duration) {
//        if (isShow)
//            Toast.makeText(App.INSTANCE, message, duration).show();
    }

    /**
     * 自定义显示Toast时间
     *
     * @param message
     * @param duration
     */
    public static void show(int message, int duration) {
//        if (isShow)
//            Toast.makeText(App.INSTANCE, message, duration).show();
    }

}

File: libbase/src/main/java/com/uncle2000/libutils/T.kt
package com.uncle2000.libutils

import android.os.Looper
import android.view.Gravity
import android.view.ViewGroup
import android.widget.TextView
import android.widget.Toast
import com.uncle2000.libbase.App


/**
 * Toast统一管理类
 */
class T private constructor() {

    init {
        /* cannot be instantiated */
        throw UnsupportedOperationException("cannot be instantiated")
    }

    companion object {

        var isShow = true

        /**
         * 短时间显示Toast
         *
         * @param message
         */
        fun showShort(message: CharSequence) {
            try {
                if (isShow && App.getInstance() != null) {

                    val toast = Toast.makeText(App.getInstance(), message, Toast.LENGTH_SHORT)
                    //处理换行文字居中问题
                    try {
                        val view = toast.view as ViewGroup
                        if (view.childCount > 0) {
                            val textView = view.getChildAt(0) as TextView
                            textView.gravity = Gravity.CENTER
                        }

                    } catch (e: Exception) {
                        e.printStackTrace()
                    }

                    toast.show()
                }
            } catch (e: Exception) {
                //解决在子线程中调用Toast的异常情况处理
                Looper.prepare()
                Toast.makeText(App.getInstance(), message, Toast.LENGTH_SHORT).show()
                Looper.loop()
            }

        }

        /**
         * 短时间显示Toast
         *
         * @param message
         */
        fun showShort(message: Int) {
            if (isShow)
                Toast.makeText(App.getInstance(), message, Toast.LENGTH_SHORT).show()
        }

        /**
         * 长时间显示Toast
         *
         * @param message
         */
        fun showLong(message: CharSequence) {
            if (isShow)
                Toast.makeText(App.getInstance(), message, Toast.LENGTH_LONG).show()
        }

        /**
         * 长时间显示Toast
         *
         * @param message
         */
        fun showLong(message: Int) {
            if (isShow)
                Toast.makeText(App.getInstance(), message, Toast.LENGTH_LONG).show()
        }

        /**
         * 自定义显示Toast时间
         *
         * @param message
         * @param duration
         */
        fun show(message: CharSequence, duration: Int) {
            if (isShow)
                Toast.makeText(App.getInstance(), message, duration).show()
        }

        /**
         * 自定义显示Toast时间
         *
         * @param message
         * @param duration
         */
        fun show(message: Int, duration: Int) {
            if (isShow)
                Toast.makeText(App.getInstance(), message, duration).show()
        }
    }

}